inc/options/common/top_nav_bar.php

```diff
@@ -0,0 +1,122 @@
+						'<sup>',
+						'</sup>',
+						'</span>'
+					),
+					array(
+						'span' => array(),
+						'sup'  => array(),
+					)
+				);
+				?>
+			</h2>
+		</a>
+		<div id="w3tc-top-nav-bar-content-links">
+			<a class="w3tc-top-nav-support" href="<?php echo Util_UI::admin_url( 'admin.php?page=w3tc_dashboard' ); ?>" alt="<?php esc_attr_e( 'Dashboard', 'w3-total-cache' ); ?>">
+				<?php esc_html_e( 'Dashboard', 'w3-total-cache' ); ?>
+			</a>
+			<div class="w3tc-top-nav-dropdown">
+				<a class="w3tc-top-nav-settings" href="<?php echo Util_UI::admin_url( 'admin.php?page=w3tc_general' ); ?>" alt="<?php esc_attr_e( 'Settings', 'w3-total-cache' ); ?>">
+					<?php esc_html_e( 'Settings', 'w3-total-cache' ); ?><span class="dashicons dashicons-arrow-down-alt2"></span>
+				</a>
+				<div id="w3tc-top-nav-settings-menu" class="w3tc-top-nav-dropdown-content">
+					<?php
+					foreach ( $settings_menu_array as $url => $data ) {
+						$target = ! empty( $data['target'] ) ? ' target="' . $data['target'] . '"': '';
+						$external = parse_url( $url, PHP_URL_HOST ) !== $_SERVER['HTTP_HOST'] ? '<span class="dashicons dashicons-external"></span>': '';
+						echo '<a href="' . $url . '" alt="' . $data['text'] . '"' . $target . '>' . $data['text'] . $external . '</a>';
+					}
+					?>
+				</div>
+			</div>
+			<div class="w3tc-top-nav-dropdown">
+				<a class="w3tc-top-nav-tools" href="<?php echo Util_UI::admin_url( 'admin.php?page=w3tc_extensions' ); ?>" alt="<?php esc_attr_e( 'Tools', 'w3-total-cache' ); ?>">
+					<?php esc_html_e( 'Tools', 'w3-total-cache' ); ?><span class="dashicons dashicons-arrow-down-alt2"></span>
+				</a>
+				<div id="w3tc-top-nav-tools-menu" class="w3tc-top-nav-dropdown-content">
+					<?php
+					foreach ( $tools_menu_array as $url => $data ) {
+						$target = ! empty( $data['target'] ) ? ' target="' . $data['target'] . '"': '';
+						$external = parse_url( $url, PHP_URL_HOST ) !== $_SERVER['HTTP_HOST'] ? '<span class="dashicons dashicons-external"></span>': '';
+						echo '<a href="' . $url . '" alt="' . $data['text'] . '"' . $target . '>' . $data['text'] . $external . '</a>';
+					}
+					?>
+				</div>
+			</div>
+			<div class="w3tc-top-nav-dropdown">
+				<a class="w3tc-top-nav-about" href="<?php echo Util_UI::admin_url( 'admin.php?page=w3tc_about' ); ?>" alt="<?php esc_attr_e( 'About', 'w3-total-cache' ); ?>">
+					<?php esc_html_e( 'About', 'w3-total-cache' ); ?><span class="dashicons dashicons-arrow-down-alt2"></span>
+				</a>
+				<div id="w3tc-top-nav-about-menu" class="w3tc-top-nav-dropdown-content">
+					<?php
+					foreach ( $about_menu_array as $url => $data ) {
+						$target   = ! empty( $data['target'] ) ? ' target="' . $data['target'] . '"': '';
+						$external = parse_url( $url, PHP_URL_HOST ) !== $_SERVER['HTTP_HOST'] ? '<span class="dashicons dashicons-external"></span>': '';
+						echo '<a href="' . $url . '" alt="' . $data['text'] . '"' . $target . '>' . $data['text'] . $external . '</a>';
```

Comment:
Escape later rather than earlier.  Escape URLs with `Util_UI::admin_url()`, target with `esc_attr()`, and the text can be translated earlier with `__()` above and then here with `esc_html()`.